README.md

```diff
@@ -166,40 +170,248 @@ Remove a key (and subsequent value) from this object. For example, to delete the
 "year" property from our car:
 
 ``` handlebars
-<button onclick={{car.delete "year"}}>
+<button onclick={{action car.delete "year"}}>
   Remove Year
 </button>
 {{!clicking will result in {make: 'Ford', model: 'Mustang'}}}
 ```
 
 #### `put(key, value)`
-#### `set(value)`
+
+Add property with a given name and value to the object. It will update the property with the given value if it already exists.
+
+```handlebars
+<button onclick={{action car.put "color" "blue"}}>
+  Add color
+</button>
+```
+
+#### `set(obj)`
+
+Replace current object microstate with a new object microstate from given hash.
+
+```handlebars
+<button onclick={{action car.set (hash make="Toyota" model="Supra" year="1982")}}>
+  Update car
+</button>
+```
 
 ### `List`
 
+List microstate represents an indexed array of values. 
```

Comment:
I don't think we have an API currently for accessing values by index do we? I'd say the main thing about lists is that they're an ordered collection.